ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -84,6 +85,10 @@ def _llmobs_set_tags(
 
             is_workflow = LLMObs._integration_is_enabled(llmobs_integration)
 
+        if kwargs.get("is_workflow") is not None:
+            # when the traced function forces an is_workflow state
+            is_workflow = bool(kwargs.get("is_workflow"))
+
```

Comment:
This fails tests because we sometimes override `kwargs` to be a list ([ref](https://github.com/DataDog/dd-trace-py/blob/2abc58a2971610cf2625a1934ac54d3da3c9814d/ddtrace/contrib/internal/langchain/patch.py#L688)). I re-discovered this by using the debugger and seeing that `kwargs` is a list. Two things we could do:
1. change the check to `if isinstance(kwargs, dict) and kwargs.get(...) is not None:`
2. pass in `is_workflow` separately from `kwargs`, which I would argue for. I see in the patching code we do `kwargs["is_workflow"] = True`, can we just pass in `is_workflow` as a keyword arg to `llmobs_set_tags` instead?

I'm personally in favor of the second

Reply:
For the second are we okay with adding it even in the base integration template ?

Reply:
ah, right. maybe there's another way we can check this? ie, a property on the span, like resource name, interface_type, etc? I think that could be a better way of doing it, instead of modifying kwargs. If we need to modify the llmobs_set_tags to take in optional kwargs, I don't think it's the worst thing, but if we can do it deterministically based on span properties first I think that's best. lmk what you think!

Reply:
I believe for now and for the sake of getting stuff shipped i'll stick to 1. I tried it and it works well. At the moment the only use case we have of force setting a span to a workflow is in the retrieval case. Maybe if we use it more then we can start thinking of establishing a more permanent spot for it.

Reply:
I think that should be OK for now, especially since we're modifying `kwargs` _after_ the call is made. But I think we should follow up with a better solution if we can, we shouldn't try to modify inputs to a function unless we're wrapping one of the arguments (ie, some kind of function)